WordPress/WordPressShareExtension/ShareViewController.swift

```diff
@@ -243,66 +277,165 @@ private extension ShareViewController {
+        let uploadPostOp = UploadOperation(context: managedContext)
+        uploadPostOp.updateWithPost(remote: remotePost)
+        uploadPostOp.groupID = groupIdentifier
+        uploadPostOp.created = NSDate()
+        uploadPostOp.currentStatus = status
+        coreDataStack.saveContext()
+        return uploadPostOp.objectID
+    }
+
+    func updatePostOperation(status: UploadOperation.UploadStatus, remotePostID: Int64, forUploadOpWithObjectID uploadOpObjectID: NSManagedObjectID) {
+        guard let uploadPostOp = (try? managedContext.existingObject(with: uploadOpObjectID)) as? UploadOperation else {
+            DDLogError("Error loading UploadOperation Object with ID: \(uploadOpObjectID)")
+            return
+        }
+        uploadPostOp.currentStatus = status
+        uploadPostOp.remotePostID = remotePostID
+        coreDataStack.saveContext()
+    }
+
+    func updateStatus(_ status: UploadOperation.UploadStatus, forUploadOpWithObjectID uploadOpObjectID: NSManagedObjectID) {
+        var uploadOp: UploadOperation?
+        do {
+            uploadOp = try managedContext.existingObject(with: uploadOpObjectID) as? UploadOperation
+        } catch {
+            DDLogError("Error loading UploadOperation Object with ID: \(uploadOpObjectID)")
+            return
+        }
+        uploadOp?.currentStatus = status
+        coreDataStack.saveContext()
+    }
+
+    func updateTaskID(_ taskID: NSNumber, forUploadOpWithObjectID uploadOpObjectID: NSManagedObjectID) {
+        var uploadOp: UploadOperation?
+        do {
+            uploadOp = try managedContext.existingObject(with: uploadOpObjectID) as? UploadOperation
+        } catch {
+            DDLogError("Error loading UploadOperation Object with ID: \(uploadOpObjectID)")
+            return
+        }
+        uploadOp?.backgroundSessionTaskID = taskID.int32Value
+        coreDataStack.saveContext()
+    }
+
+    func fetchPostUploadOp(withObjectID uploadOpObjectID: NSManagedObjectID) -> UploadOperation? {
+        var uploadOp: UploadOperation?
+        do {
+            uploadOp = try managedContext.existingObject(with: uploadOpObjectID) as? UploadOperation
+        } catch {
+            DDLogError("Error loading UploadOperation Object with ID: \(uploadOpObjectID)")
+        }
+        return uploadOp
+    }
+
+    func fetchMediaUploadOpsForGroup(_ groupID: String) -> [UploadOperation]? {
+        var mediaUploadOps: [UploadOperation]
+        let request = NSFetchRequest<NSFetchRequestResult>(entityName: "UploadOperation")
+        request.predicate = NSPredicate(format: "(groupID == %@ AND isMedia == true)", groupID)
+        do {
+            mediaUploadOps = try managedContext.fetch(request) as! [UploadOperation]
+        } catch {
+            fatalError("Failed to fetch UploadOperations: \(error)")
+        }
+
+        return mediaUploadOps
+    }
+}
+
+// MARK: - ShareViewController Extension: Backend Interaction
+
 private extension ShareViewController {
```

Comment:
These methods on this extension could also move to the ShareExtensionStore class.